pkg/store/configuration.go

```diff
@@ -15,25 +15,32 @@
 package store
 
 import (
+	"fmt"
 	"github.com/onosproject/onos-config/pkg/store/change"
+	"github.com/openconfig/gnmi/proto/gnmi"
+	"regexp"
 	"sort"
 	"strings"
 	"time"
 )
 
+const configurationNamePattern = `[a-zA-Z0-9\-:_]{4,40}`
+const configurationVersionPattern = `[a-zA-Z0-9_\.]{2,10}`
```

Comment:
are we happy with these name restrictions for version and devicetype (max 10 chars)

Reply:
I think 10 for version is good and 40 for name should be fine.

Reply:
+1